apps/explorer/lib/explorer/exchange_rates/source/mobula.ex

```diff
@@ -0,0 +1,171 @@
+        case Explorer.Chain.Hash.Address.cast(address_hash_string) do
+          {:ok, address_hash} ->
+            acc
+            |> Map.put(address_hash, %{
+              fiat_value: Map.get(market_data, "price"),
+              circulating_market_cap: Map.get(market_data, "market_cap"),
+              volume_24h: Map.get(market_data, "volume")
+            })
+
+          _ ->
+            acc
+        end
+
+      _, acc ->
+        acc
+    end)
+  end
+
+  @impl Source
+  def format_data(_), do: []
+
+  @impl Source
+  def source_url do
+    "#{base_url()}/market/data?asset=#{Explorer.coin()}"
+  end
+
+  @impl Source
+  def source_url(token_addresses) when is_list(token_addresses) do
+    joined_addresses = token_addresses |> Enum.map_join(",", &to_string/1)
+
+    "#{base_url()}/market/multi-data?blockchains=#{chain()}&assets=#{joined_addresses}"
+  end
+
+  @impl Source
+  def source_url(input) do
+    symbol = input
+    "#{base_url()}/market/data&asset=#{symbol}"
+  end
+
+  def history_source_url do
+    id = config(:secondary_coin_id)
+
+    if id, do: "#{base_url()}/market/history?asset=#{id}", else: "#{base_url()}/market/history?asset=#{Explorer.coin()}"
+  end
+
+  @spec history_url(non_neg_integer()) :: String.t()
+  def history_url(previous_days) do
+    now = DateTime.utc_now()
+    date_days_ago = DateTime.add(now, -previous_days, :day)
+    timestamp_ms = DateTime.to_unix(date_days_ago) * 1000
+
+    Logger.info(
+      "Fetching price history from Mobula: #{history_source_url()} for the last #{previous_days} days. Timestamp: #{timestamp_ms} ms. Now: #{DateTime.to_unix(now)} ms. Date days ago: #{DateTime.to_unix(date_days_ago)} ms."
+    )
+
+    "#{history_source_url()}&from=#{timestamp_ms}"
+  end
+
+  @impl Source
+  def headers do
+    if api_key() do
+      [{"Authorization", "#{api_key()}"}]
+    else
+      []
+    end
+  end
+
+  defp api_key do
+    config(:api_key) || nil
+  end
+
+  defp get_current_price(market_data) do
+    if market_data["price"] do
+      to_decimal(market_data["price"])
+    else
+      1
+    end
+  end
+
+  defp get_btc_value(id, market_data) do
+    case get_btc_price() do
+      {:ok, price} ->
+        btc_price = to_decimal(price)
+        current_price = get_current_price(market_data)
+
+        if id != "btc" && current_price && btc_price do
+          Decimal.div(current_price, btc_price)
+        else
+          1
+        end
+
+      _ ->
+        1
+    end
+  end
+
+  defp chain do
+    config(:chain_id) || "ethereum"
+  end
+
+  defp base_url do
+    config(:base_url) || "https://api.mobula.io/api/1"
+  end
+
+  defp get_btc_price do
+    url = "#{base_url()}/market/data?asset=Bitcoin"
+
+    case Source.http_request(url, headers()) do
+      {:ok, data} = resp ->
+        if is_map(data) do
+          current_price = data[~c"price"]
+
+          {:ok, current_price}
+        else
+          resp
+        end
+
+      resp ->
+        resp
+    end
```

Comment:
nitpick: I might be missing something, but can't we rewrite it like this? Just to lower cyclomatic complexity.

```suggestion
    Source.http_request(url, headers())
    |> case do
      {:ok, %{"price" => current_price}} ->
        {:ok, current_price}
    
      resp ->
        resp
    end
```